plugin/trino-redshift/src/main/java/io/trino/plugin/redshift/RedshiftClient.java

```diff
@@ -666,10 +665,9 @@ public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connect
                         RedshiftClient::writeChar));
 
             case Types.VARCHAR: {
-                if (type.columnSize().isEmpty()) {
-                    throw new TrinoException(REDSHIFT_INVALID_TYPE, "column size not present");
-                }
-                int length = type.requiredColumnSize();
+                // Redshift column exposes precision with max precision upto `REDSHIFT_MAX_VARCHAR`.
+                // Defaulting to `VarcharType.MAX_LENGTH`(instead of `REDSHIFT_MAX_VARCHAR`) as synthetic column created by Trino creates unbounded varchar.
+                int length = type.columnSize().orElse(VarcharType.MAX_LENGTH);
```

Comment:
But do we need to set it to `MAX_LENGTH` i.e if Redshift doesn't have a unbounded varchar type - should we allow cast pushdown for such operations ? Like what if we support some concat operation in future and if the concat operation results in data length more than Redshift max length ? Trino would execute such queries but redshift might fail for such queries.

Reply:
I removed change in this class by not supporting unbounded varchar.